### PR TITLE
[AKS] Allow updating the ssh key value if cluster was created without ssh key

### PR DESCRIPTION
**Pull request by bingosummer**
If the cluster was created with `az aks create --name $CLUSTER_NAME --resource-group $RG_NAME --no-ssh-key`, this PR allows the customer to update ssh public key with `az aks update --name $CLUSTER_NAME --resource-group $RG_NAME --ssh-key-value $YOUR_SSH_KEY_VAULE`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -2793,14 +2793,24 @@ def update_linux_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         if ssh_key_value:
             if mc.linux_profile is None:
-                raise InvalidArgumentValueError("Updating the cluster with no ssh key set at creation is not supported")
-            mc.linux_profile.ssh = self.models.ContainerServiceSshConfiguration(
-                public_keys=[
-                    self.models.ContainerServiceSshPublicKey(
-                        key_data=ssh_key_value
+                mc.linux_profile = self.models.ContainerServiceLinuxProfile(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2796]: Will rp handle the backfill if admin_username not set previsouly? If so, may skip the backfill here

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -2793,14 +2793,24 @@ def update_linux_profile(self, mc: ManagedCluster) -> ManagedCluster:
...
+                            self.models.ContainerServiceSshPublicKey(
+                                key_data=ssh_key_value
+                            )
+                        ]
                     )
-                ]
-            )
+                )
+            else:
+                mc.linux_profile.ssh = self.models.ContainerServiceSshConfiguration(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2807]: still need to check if .ssh is None or not?

**Review by FumingZhang — COMMENTED** [review]
Better add a new live test case (or update existing one) to test the change

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
Comment (bingosummer) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2796]: If I don't set "admin_username" here, I'll get an error like:

```
  File "/home/binxi/.azure/cliextensions/aks-preview/azext_aks_preview/managed_cluster_decorator.py", line 2796, in update_linux_profile
    mc.linux_profile = self.models.ContainerServiceLinuxProfile(
TypeError: ContainerServiceLinuxProfile.__init__() missing 1 required keyword-only argument: 'admin_username'
```

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 2 above)
Comment (bingosummer) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2807]: no matter it's none or not, we need to update `.ssh`.

**Comment by bingosummer** [comment]
> Better add a new live test case (or update existing one) to test the change

added.

**Comment by bingosummer** [comment]
Queued a [live test pipeline](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=69211902&view=results) to verify the change

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**